### ISSUE TITLE
v0.24 - potentially cause of "[homeassistant] Error doing job: socket.accept() out of system resource" 

### ISSUE THREAD

**Issue by essenemari** [macxq/foxess-ha#156]
Since 2023.4.4 up to 2023.5.2 HA Core (rpi4) with FoxEss integration v0.24, v0.25, v0.26 causes Home Assistant failure with error message in HA log as below:

```
2023-04-12 09:06:04.403 ERROR (MainThread) [homeassistant] Error doing job: socket.accept() out of system resource
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 159, in _accept_connection
  File "/usr/local/lib/python3.10/socket.py", line 293, in accept
OSError: [Errno 24] Too many open files
```

Once error occurred in the HA log, the HA log was rapidly increasing  8MB/60s. 
Home Assistant stopped responding.  
Needed to restart HA to back to normal.

Looks like the Foxess integration cause failure. 
Just before a failure, the counter below always shows the 1400  (+/- 5) entries:
```
lsof -u homeassistant | wc -l
```

The above increases because of following connections:
```
hass    1379 homeassistant  587u     IPv4            1864418       0t0     TCP rpi4:34774->8.209.116.72:https (CLOSE_WAIT)
hass    1379 homeassistant  588u     IPv4            1889086       0t0     TCP rpi4:55170->8.209.116.72:https (CLOSE_WAIT)
hass    1379 homeassistant  589u     IPv4            1816663       0t0     TCP rpi4:35416->8.209.116.72:https (CLOSE_WAIT)
hass    1379 homeassistant  590u     IPv4            1903074       0t0     TCP rpi4:32842->8.209.116.72:https (CLOSE_WAIT)
hass    1379 homeassistant  591u     IPv4            1839019       0t0     TCP rpi4:40650->8.209.116.72:https (CLOSE_WAIT)
hass    1379 homeassistant  592u     IPv4            1875714       0t0     TCP rpi4:38744->8.209.116.72:https (CLOSE_WAIT)
```

8.209.116.72 - FoxEss's IP address.

Just before the failures, the proportions are as following
```
lsof for homeassistant user: 1400
lsof for homeassistant user and  8.209.116.72: 973

lsof for homeassistant user: 1403
lsof for homeassistant user and  8.209.116.72: 985

lsof for homeassistant user: 1403
lsof for homeassistant user and  8.209.116.72: 983

lsof for homeassistant user: 1403
lsof for homeassistant user and  8.209.116.72: 982

lsof for homeassistant user: 1403
lsof for homeassistant user and  8.209.116.72: 980

lsof for homeassistant user: 1403
lsof for homeassistant user and  8.209.116.72: 990

lsof for homeassistant user: 1403
lsof for homeassistant user and  8.209.116.72: 987

lsof for homeassistant user: 1403
lsof for homeassistant user and  8.209.116.72: 983
```

**Expected behavior**
Expected less connections to FoxEss cloud over the integration (8.209.116.72).  With so many such connections, it is very close to exhaust available sockets and hit a HA serious crash (drain disk space because HA log increase (8MB/min)).

**HA version:**
2023.4.4 - 2023.5.2

**foxess-ha version:**
v0.24, v0.25, v0.26

**Back to operational state**
Restart HA helps as workaround till next limit drain, usually it is next day.

**Workaround**
Remove the integration.

**Comment by essenemari** [comment]
Issue persist on my end. More or less one failure per 24h. 

19 hours after last failure here here counters for FoxEess IP:

```
root@rpi4:/home/art# lsof -w | grep "TCP\|UDP" | grep 8.209.116.72|wc -l
11267
root@rpi4:/home/art# lsof -w | grep "TCP\|UDP" | grep CLOSE_WAIT | grep 8.209.116.72|wc -l
11191
```
...and increasing till failure. Is it correct behavior?

**Comment by essenemari** [comment]
fixed with 0.30b

**Comment by essenemari** [comment]
appeared again with v0.31
